### PR TITLE
libidset: replace internal nodeset class

### PR DESCRIPTION
**Pull request by garlick**
The old "nodeset" class has some deficiencies documented in #1495.  We created `libflux-idset` as an external, standalone library to export nodeset-like functionality to framework projects, but only created the bare minimum needed by flux-sched at the time and left nodeset in place.

In the spriit of recent API cleanup efforts, this PR finishes the job, adding functionality to idset, converting users, and removing nodeset.

### CODE REVIEW

**Comment by garlick** [comment]
Still some squashing to do here before a review.  Sorry for the noise.

**Comment by garlick** [comment]
Rebased.

One builder fail the valgrind test:

```
==963== 6,310,202 (224 direct, 6,309,978 indirect) bytes in 1 blocks are definitely lost in loss record 103 of 103
==963==    at 0x4C31D2F: realloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==963==    by 0x4E87BF9: ev_realloc (ev.c:1741)
==963==    by 0x4E881E4: ev_feed_event (ev.c:2005)
==963==    by 0x4E8B12F: queue_events (ev.c:2034)
==963==    by 0x4E8B12F: ev_run (ev.c:3714)
==963==    by 0x4E585C2: flux_reactor_run (reactor.c:140)
==963==    by 0xB7C910F: mod_main (job.c:938)
==963==    by 0x1142EB: module_thread (module.c:158)
==963==    by 0x55BA6DA: start_thread (pthread_create.c:463)
==963==    by 0x636188E: clone (clone.S:95)
==963== 
```

looks unrelated so I restarted it.

**Comment by grondo** [comment]
Two quick comments, neither should be considered must-fix:

 * 73f7634: it would be slightly cleaner to add libidset to libflux_internal in a separate commit. Lumping it in with the mrpc update seems arbitrary and isn't symmetrical with the other code that is updated. (or maybe it is? I would also accept a comment in the commit message that describes why libidset was moved into libflux_internal in *this* commit)

 * e6b2e84 could use a commit message. The commit subject says the linkage was "fixed" but there isn't a hint why (which could be helpful for future developer wondering why that test cant use `$(test_ldadd)` like all the other tests)

**Comment by garlick** [comment]
Thanks, good suggestions!

**Comment by garlick** [comment]
OK, I think I addressed @grondo's comments.  I also expanded some commit messages a bit.

**Comment by garlick** [comment]
Let me see if I can get the idset coverage up just a bit before this gets merged.

I also feel like the way everything is being linked is a little convoluted and might be improved if the veb stuff moved to libidset (though I've tried that twice now and got myself in the weeds both times).

**Comment by garlick** [comment]
Rebased on current master and squashed down the unit test addition.  I think this could be ready to go in.  @chu11, sounds like @grondo may be offline for a bit.  Would you mind giving this a quick review?

File: src/common/libidset/idset.h [hunk 1]
```diff
@@ -57,10 +65,44 @@ char *idset_encode (const struct idset *idset, int flags);
...
 struct idset *idset_decode (const char *s);
 
-/* Make id a member of idset.
+/* Add id (or range [lo-hi]) to idset.
  * Return 0 on success, -1 on failure with errno set.
  */
 int idset_set (struct idset *idset, unsigned int id);
+int idset_range_set (struct idset *idset, unsigned int lo, unsigned int hi);
+
+/* Remove id (or range [lo-hi] from idset.
```
Comment (chu11) [inline src/common/libidset/idset.h]: stupid typo, no end paren ')'

File: src/common/libidset/idset.c [hunk 2]
```diff
@@ -124,6 +157,124 @@ int idset_set (struct idset *idset, unsigned int id)
...
+}
+
+bool idset_compare (const struct idset *idset1,
+                    const struct idset *idset2)
+{
+    unsigned int id;
+
+    if (!idset1 || !idset2)
+        return false;
+
```
Comment (chu11) [inline src/common/libidset/idset.c]: minor thought, would it be a net win to do some obvious checks before iterating through everything below?  I know ```vebsize()``` is costly and perhaps not worth it.  T.M != T.M?  I don't know the library that well, so maybe it's not a net win.

File: src/common/libidset/test/idset.c [hunk 3]
```diff
@@ -154,6 +413,12 @@ int main (int argc, char *argv[])
     test_badparam ();
     test_codec ();
     test_codec_large ();
+    test_iter ();
+    test_set ();
+    test_range_set ();
+    test_clear ();
+    test_range_clear ();
+    test_copy ();
```
Comment (chu11) [inline src/common/libidset/test/idset.c]: i don't think i see a unit test for AUTOGROW?

Comment (garlick) [reply src/common/libidset/test/idset.c]: Thanks will add some.

File: src/common/libidset/idset.c (hunk 2 above)
Comment (garlick) [reply src/common/libidset/idset.c]: `T.M` is the number of available slots in the set, not filled slots.  `vebsize()` is the size in bytes of the data structure, not related to filled slots.  So I don't think either of those is helpful here.  

In looking at this, I realized there are no users of `idset_compare()` so hey, let's just get rid of it.

**Comment by garlick** [comment]
LMK how this looks and I'll squash the incremental devel.

File: src/common/libidset/test/idset.c [hunk 4]
```diff
@@ -275,7 +275,7 @@ void test_range_set (void)
 
     ok (idset_range_set (idset, 0, 2) == 0,
         "idset_range_set 0-2 worked");
-    ok (idset_range_set (idset, 80, 79) == 0,
+    ok (idset_range_set (idset, 80, 79) == 0, // revsersed
```
Comment (chu11) [inline src/common/libidset/test/idset.c]: typo "revsersed"

Comment (garlick) [reply src/common/libidset/test/idset.c]: Sheesh, I blame this keyboard.

**Comment by garlick** [comment]
Fixed that typo and squashed.

**Comment by chu11** [comment]
lgtm, I'll hit the button once travis passes.  You'll have to rebase this or #1867, depending on what finishes first :-)

**Comment by garlick** [comment]
Thanks!

**Pull request merged**